sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -140,22 +140,24 @@ def fit(self, X, y, sample_weight=None):
                                             dtype='u8')
 
         self._validate_parameters()
-        n_samples, self.n_features_ = X.shape  # used for validation in predict
+
+        # used for validation in predict
+        n_samples, self._n_features = X.shape
 
         # we need this stateful variable to tell raw_predict() that it was
         # called from fit() (this current method), and that the data it has
         # received is pre-binned.
         # predicting is faster on pre-binned data, so we want early stopping
-        # predictions to be made on pre-binned data. Unfortunately the scorer_
+        # predictions to be made on pre-binned data. Unfortunately the _scorer_
         # can only call predict() or predict_proba(), not raw_predict(), and
         # there's no way to tell the scorer that it needs to predict binned
         # data.
         self._in_fit = True
 
         if isinstance(self.loss, str):
-            self.loss_ = self._get_loss(sample_weight=sample_weight)
+            self._loss_ = self._get_loss(sample_weight=sample_weight)
```

Comment:
Uhm we need second advice but I think we don't need the second underscore when we have a leading underscore (as with `self._n_feature`. And this will be the case for all the variables renamed here.

Reply:
Ah sorry, this was probably meant by @NicolasHug too... I didn't understand correctly.